pkg/kube/client.go

```diff
@@ -977,17 +986,23 @@ func isFoundPod(podItem []v1.Pod, pod v1.Pod) bool {
 	return false
 }
 
-func asVersioned(info *resource.Info) runtime.Object {
+func asVersionedOrUnstructured(info *resource.Info) runtime.Object {
+	obj, _ := asVersioned(info)
```

Comment:
Would it be better to just have the caller ignore the error instead of creating a separate function?

Reply:
@thomastaylor312 Hey! Thanks a lot for your review 😄 

I thought so at first but ended up with the separate functions for readability and reliability reasons.

The way `asVersioned` works is that it returns an unstructured object rather than a versioned object, which imho can't be inferred from its name. Theoretically I could have given `asVersioned` an another name or make the implementation simpler so that naming it can be also easier. But I was unable to come up with a great idea and had no confidence to not break it along the way.

With these separate functions, anyway, they are easy to read and I'm sure it doesn't break.

`asVersioned` just returns an error when the object isn't versioned(actually it returns an unstructured object with an error object if it wasn't registered as a versioned object but you don't usually use the object in that case).

`asVersionedOrUnstructured` does what its name tells - returns either a versioned or an unstructured object.

That said, I just wanted to clarify my intention here. I'm still open to suggestions. 

So, If you still think the separate function isn't necessary, I'm absolutely fine to merge the functions into one and ignore errors on caller-side when necessary.

WDYT?

Reply:
I trust your opinion on this, so I am fine with having the separate functions. I'll be porting these over to Helm 3 as well (where they are actually exported functions), so now is as good of time as any to change up the function signatures anyway